rock_build_autoproj_2

```diff
@@ -0,0 +1,450 @@
+    AUTOPROJ_OSDEPS_MODE=all
+fi
+
+if test "x$SKIP_SUCCESSFUL" = "xtrue" && test -d dev; then
+    if test -f dev/successful; then
+        if test "x$DOCGEN" != "xtrue" || test -f dev/doc-successful; then
+            echo "last build was successful and SKIP_SUCCESSFUL is set, doing nothing"
+            exit 0
+        else
+            MODE=incremental
+        fi
+    fi
+fi
+
+if test -z "$RUBY"; then
+    RUBY=ruby
+fi
+if test -z "$GEM"; then
+    GEM=gem
+fi
+if test -z "$MODE"; then
+    MODE=auto
+fi
+
+
+if test "x$MODE" = "xincremental"; then
+    echo "MODE=incremental, doing an incremental build"
+    do_incremental=1
+    do_full_cleanup=0
+elif test "x$MODE" = "xfull-bootstrap"; then
+    echo "MODE=full-bootstrap, doing a full build, including checking out packages"
+    do_incremental=0
+    do_full_cleanup=1
+elif test "x$MODE" = "xbootstrap"; then
+    echo "MODE=bootstrap, doing a full build but not checking out packages again. Use full-bootstrap for that"
+    do_incremental=0
+    do_full_cleanup=0
+elif test -d dev && ! test -f dev/successful; then
+    echo "last build was unsuccessful, doing an incremental build"
+    do_incremental=1
+    do_full_cleanup=0
+else
+    echo "doing a full build"
+    do_incremental=0
+    do_full_cleanup=0
+fi
+
+if test "x$do_incremental" = "x1" && test -f dev/cleaned; then
+    echo "this is an incremental build, but the last build got cleaned, doing nothing"
+    exit 0
+fi
+
+rm -f dev/successful
+rm -f dev/doc-successful
+rm -f dev/cleaned
+rm -f docgen.txt
+rm -rf api
+
+# If the current dev/install folder has a cache of the downloaded archives, then
+# save it
+if test -d dev/install/cache; then
+    mkdir -p archive_cache
+    rsync -a dev/install/cache/ archive_cache/
+fi
+
+if test "x$do_incremental" = "x0"; then
+    if test "x$do_full_cleanup" = "x1"; then
+        rm -rf dev
+    elif test -d dev; then
+        rm -rf dev/install
+        find dev -type d -name build -exec rm -rf {} \; -prune
+        rm -rf dev/autoproj
+#        bundle update
+    fi
+fi
+
+# Always delete autoproj's configuration directory so that we always start fresh
+# Otherwise, it would not get updated
+rm -rf dev/.autoproj/remotes
+
+# If there is an archive cache, copy it into our working directory
+if test -d archive_cache; then
+    mkdir -p dev/install/cache
+    rsync -a archive_cache/ dev/install/cache/
+fi
+
+
+export AUTOPROJ_OSDEPS_MODE
+export AUTOPROJ_BOOTSTRAP_IGNORE_NONEMPTY_DIR=1
+
+# Install yard early, some C++ packages with Ruby bindings will pick it up and
+# use it to generate the Ruby binding documentation
+#if test "x$DOCGEN" = "xtrue"; then
+#    $GEM install rdoc yard --no-rdoc --no-ri
+#fi
+
+if test "x$DOCGEN_ONLY" = "xtrue"; then
+    do_doc_only=1
+    DOCGEN="true"
+else
+    do_doc_only=0
+fi
+
+if test "x$RUN_TESTS" = "xtrue"; then
+    BUILD_TESTS=true
+fi
+
+# If there are no autoproj/ directory, we need to start by bootstrapping
```

Comment:
This test is about the existence of a buildconf folder aka "autoproj/" dev is the main checkout folder.

This part uses autoproj_bootstrap to create a new bootstrap in case no buildconf (dev/autoproj folder) is already present.

--seed-config could replace line 287, where is an actual bug, ad this should only happen on new bootstraps